packages/react/src/components/form/form.tsx

```diff
@@ -14,42 +14,134 @@ import {
   ConfigurationOverrides,
   ConfigurationOverridesProps,
 } from "../configuration-overrides";
-import { LoginOptions } from "../../domain/types";
+import { Handle, HandleType, LoginOptions } from "../../domain/types";
+import React, { useCallback } from "react";
+import { Slots, useSlots } from "../slot";
+import { Factor } from "@slashid/slashid";
+import { Flag } from "../input";
 
 export type Props = ConfigurationOverridesProps & {
   className?: string;
   onSuccess?: CreateFlowOptions["onSuccess"];
   middleware?: LoginOptions["middleware"];
+  children?: Slots<
+    "initial" | "authenticating" | "success" | "error" | "footer"
+  >; // TS does not enforce this, but it is used for documentation
```

Comment:
@ikovic if you think it could be helpful, we could at least enforce the slot name by exporting a specific `<FormSlot>` component instead of the generic `<Slot>` that will enforce the name type like this:
```
type FormSlotName = "initial" | "authenticating" | "success" | "error" | "footer";

export function FormSlot({ children, name }: SlotProps<FormSlotName>) {
  return <Slot name={name}>{children}</Slot>;
}
```

Reply:
This would break the implementation - only a `Slot` instance can be rendered as an immediate child of the parent container.

Reply:
I feel quite strongly that we should keep `<Slot>` as the top-level API so we can build some raport with consumers, and grow their understanding of this concept.

That said maybe a compromise is to offer the type as an export to use as a generic? It's ugly but some might find it helpful

```ts
const Component = () => {
  return (
    <Slot<FormSlots> name="initial">
    </Slot>
  )
}
```